src/services/etag/etag.ts

```diff
@@ -0,0 +1,66 @@
+import { IService, Request, Response, Service } from "../../../mod.ts";
+import { createHash } from "./deps.ts";
+
+export class EtagService extends Service implements IService {
+  #options: { weak: boolean };
+
+  constructor(options: { weak: boolean } = { weak: false }) {
+    super();
+    this.#options = options;
+  }
+
+  runAfterResource(request: Request, response: Response) {
+    // if response body is empty, send a default etag
+    if (
+      response.body === null ||
+      (typeof response.body === "string" && response.body.length === 0)
+    ) {
+      // when it's empty, we want to set a default etag
+
+      // but if etag is already present on request, send a 304
+      if (request.headers.get("if-none-match")) {
+        response.status = 304;
+        response.body = null;
+      } else { // set the NEW default etag
+        response.headers.set("Last-Modified", new Date().toUTCString());
```

Comment:
I would prefer this to be lower case.

Reply:
https://github.com/drashland/drash/pull/568/commits/10fb43cee7e0499411f7813158354ff1953876fb + below thread